Sources/GATTCharacteristic.swift

```diff
@@ -1841,6 +1841,134 @@ extension GATTAerobicThreshold: CustomStringConvertible {
     }
 }
 
+/**
+ Alert Status
+ 
+ The Alert Status characteristic defines the Status of alert.
+ Bit 0, meaning “Ringer State”
+ Bit 1, meaning “Vibrator State”
+ Bit 2, meaning "Display Alert Status"
+ 
+ - SeeAlso: [Alert Status](https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.alert_status.xml)
+ */
+public struct GATTAlertStatus: GATTProfileCharacteristic {
+    
+    internal static let length = MemoryLayout<UInt8>.size
+    
+    public static var uuid: BluetoothUUID { return .alertStatus}
```

Comment:
you don't actually need a `struct AlertStatus`. GATTAlertStatus can contain a property states.

Something like 

`public var states: BitMaskOptionSet<State>`

and your State enum should be something like:

```
enum State {
    case Ringer = 0b01
    case Vibrate = 0b10
    case Display = 0b100
}
```

Reply:
Ok